sklearn/linear_model/ridge.py

```diff
@@ -301,8 +312,13 @@ def ridge_regression(X, y, alpha, sample_weight=None, solver='auto',
         if np.atleast_1d(sample_weight).ndim > 1:
             raise ValueError("Sample weights must be 1D array or scalar")
 
-        # Sample weight can be implemented via a simple rescaling.
-        X, y = _rescale_data(X, y, sample_weight)
+        if solver != 'sag':
+            # Sample weight can be implemented via a simple rescaling.
+            # FIX: In the sparse case, with fit_intercept=True, we cannot
+            # center the data, so a rescaling is incorrect! See #4710.
+            # Sag solver can actually handle sample weights without rescaling,
+            # and can be used as a reference to solve #4710.
```

Comment:
Actually, as `fit_intercept` is not passed to `ridge_regression`, even sag solver cannot fit the intercept in the sparse case.
Shall I add a parameter to `ridge_regression` or always raise an error when `fit_intercept=True` in the sparse case?